### PR TITLE
Documentation: Update with explanations of Tracer methods

### LINKED ISSUE #123: Documentation: Update with explanations of Tracer methods
Update the documentation with [comments from @fbogsany](https://github.com/open-telemetry/opentelemetry-ruby/pull/118#issuecomment-542497023) during the review of #118.

### PR DESCRIPTION
**Pull request by duonoid**
Fixes #123

https://github.com/open-telemetry/opentelemetry-ruby/pull/118#issuecomment-542497023

### CODE REVIEW

**Comment by duonoid** [comment]
I signed it

**Comment by duonoid** [comment]
I signed it

**Comment by duonoid** [comment]
I signed it

**Comment by duonoid** [comment]
I signed it

File: api/lib/opentelemetry/trace/tracer.rb [hunk 1]
```diff
@@ -40,6 +46,17 @@ def start_root_span(name, attributes: nil, links: nil, start_timestamp: nil, kin
...
+      #
+      # Parent context can be either passed explicitly, or inferred from currently activated span.
+      #
+      # @param [optional Span] with_parent Explicitly managed parent Span, overrides
+      #   +with_parent_context+.
+      # @param [optional SpanContext] with_parent_context Explicitly managed. Overridden by
+      #   +with_parent+.
+      #
+      # @return [Span]
       def start_span(name, with_parent: nil, with_parent_context: nil, attributes: nil, links: nil, start_timestamp: nil, kind: nil, sampling_hint: nil)
```
Comment (fbogsany) [inline api/lib/opentelemetry/trace/tracer.rb]: Should we document all the other params here and for `in_span` above in this PR or would you rather address that separately?

Comment (duonoid) [reply api/lib/opentelemetry/trace/tracer.rb]: I don't think I have enough context yet to fully understand the other params, so "separately" would be better for me at this point.

**Pull request merged**